### PR TITLE
feat: plumb SDK test data branch through to URL used for test config

### PR DESCRIPTION
**Pull request by typotter**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -21,8 +21,8 @@ on:
         required: false
 
 env:
-  SDK_BRANCH_NAME: ${{ inputs.sdk_branch  || github.head_ref || github.ref_name }}
```
Comment (typotter) [inline .github/workflows/test.yaml:24]: cleaning up names for consistency with other repos

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -131,7 +131,8 @@ jobs:
             touch app/emulator.log                            # create log file
             chmod 777 app/emulator.log                        # allow writing to log file
             adb logcat | grep EppoSDK >> app/emulator.log &   # pipe all logcat messages into log file as a background process
-            ./gradlew connectedCheck --no-daemon              # run tests
+            ./gradlew connectedCheck --no-daemon \
+              -Pandroid.testInstrumentationRunnerArguments.TEST_DATA_BRANCH=${{ env.TEST_DATA_BRANCH }}           # run tests
```
Comment (typotter) [inline .github/workflows/test.yaml:135]: there are few options available to pass options at runtime to the `connectedCheck`. This is the simplest one I could find.

File: eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java [hunk 3]
```diff
@@ -781,7 +789,7 @@ private void waitForPopulatedCache() {
       while (!cachePopulated) {
         if (System.currentTimeMillis() > waitEnd) {
           throw new InterruptedException(
-              "Cache file never populated; assuming configuration error");
+              "Cache file never populated or smaller than expected 8000 bytes; assuming configuration error");
```
Comment (typotter) [inline eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java:792]: wasted a bunch of time trying to figure this out (my test data branch had a very small config).

Comment (felipecsl) [reply eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java:792]: where is this "expected 8000 bytes" coming from? first time I'm hearing about it I think

**Review by felipecsl — APPROVED** [review]

File: eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java (hunk 3 above)
Comment (typotter) [reply eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java:792]: There's a check below that checks for a minimum cache file size of 8k

Comment (typotter) [reply eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java:792]: [line 794/995](https://github.com/Eppo-exp/android-sdk/pull/137/files/b1c6a6bcf14f2e8b77e58869296d2bf5bf192468#diff-38172de297d3530a02668c9bbc0f4ca214d4ed7386f4e7b73b8eae0f477245b0R794) cc: @felipecsl

File: eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java [hunk 4]
```diff
@@ -22,6 +22,7 @@
 import android.util.Log;
 import androidx.annotation.Nullable;
 import androidx.test.core.app.ApplicationProvider;
+import androidx.test.platform.app.InstrumentationRegistry;
```
Comment (aarsilv) [inline eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java:25]: 👀

File: eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java [hunk 5]
```diff
@@ -68,8 +69,15 @@ public class EppoClientTest {
   private static final String TAG = logTag(EppoClient.class);
   private static final String DUMMY_API_KEY = "mock-api-key";
   private static final String DUMMY_OTHER_API_KEY = "another-mock-api-key";
-  private static final String TEST_HOST =
+
+  // Use branch if specified by env variable `TEST_DATA_BRANCH`.
+  private static final String TEST_BRANCH =
+      InstrumentationRegistry.getArguments().getString("TEST_DATA_BRANCH");
```
Comment (aarsilv) [inline eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java:75]: Cool did not know about `InstrumentRegistry`

**Review by aarsilv — APPROVED** [review]
📈

**Pull request merged**